src/NLog.Targets.AppCenter/AppCenterTarget.cs

```diff
@@ -192,10 +200,72 @@ private void TrackEvent(string eventName, Exception exception, IDictionary<strin
                 properties = properties ?? new Dictionary<string, string>(1);
                 if (properties.Count < 20)
                     properties["EventName"] = eventName;
-                Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties);
+
+                Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties, attachmentLogs);
             }
 
             Microsoft.AppCenter.Analytics.Analytics.TrackEvent(eventName, properties);
         }
+
+        /// <remarks>
+        ///     The maximun file size (not compressed) is 10mb
+        ///     The maximum attached file (compressed) size is 1mb.
+        ///     The maximum attached file count is 10.
+        /// </remarks>
+        private Microsoft.AppCenter.Crashes.ErrorAttachmentLog[] BuildAttachmentLogs(LogEventInfo logEvent)
+        {
+            var path = RenderLogEvent(PathToCrashAttachmentDirectory, logEvent);
+            if (!string.IsNullOrEmpty(path))
+            {
+                var errorAttachements = new List<Microsoft.AppCenter.Crashes.ErrorAttachmentLog>();
+                try
+                {
+                    var directoryToCompress = new System.IO.DirectoryInfo(path);
+                    if (directoryToCompress.Exists)
+                    {
+                        foreach (System.IO.FileInfo fileToCompress in directoryToCompress.GetFiles())
+                        {
+                            // 10mb file size limit
+                            if (fileToCompress.Length > 1024 * 1024 * 10)
+                                continue;
+
+                            if (errorAttachements.Count >= 10)
+                                break;
+
+                            if ((System.IO.File.GetAttributes(fileToCompress.FullName) & System.IO.FileAttributes.Hidden) != System.IO.FileAttributes.Hidden)
+                            {
+                                using (System.IO.FileStream originalFileStream = fileToCompress.OpenRead())

```

Comment:
`File.OpenRead()` ends up calling this method:
```csharp
return new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read);
```
And because of [FileShare.Read](https://docs.microsoft.com/en-us/dotnet/api/system.io.fileshare) then it will fail if the file is open for writing (ex. file is currently open for writing logs).